lms/djangoapps/learner_dashboard/tests/test_programs.py

```diff
@@ -326,10 +328,23 @@ def setUp(self):
         self.set_program_in_catalog_cache(self.program_uuid, self.program)
         self.create_programs_config()
 
+        discussion_config = ProgramDiscussionsConfiguration.objects.create(
+            program_uuid=self.program_uuid,
+            enabled=True,
+            provider_type="piazza",
+        )
+        discussion_config.lti_configuration = LtiConfiguration.objects.create(
+            config_store=LtiConfiguration.CONFIG_ON_DB,
+            lti_1p1_launch_url='http://test.url',
+            lti_1p1_client_key='test_client_key',
+            lti_1p1_client_secret='test_client_secret',
+        )
+        discussion_config.save()
+
     def test_discussion_flags_exist(self):
         """
         Test if programTabViewEnabled and discussionFragment exist in html.
         """
         response = self.client.get(self.url)
         self.assertContains(response, 'programTabViewEnabled: true',)
-        self.assertContains(response, 'discussionFragment: {"configured": false, "iframe": ""')
+        self.assertContains(response, 'discussionFragment: {"configured": true, "iframe": ""')
```

Comment:
Why has this changed? What are we testing here?

Reply:
Response `programTabViewEnabled` was false in this context because of additional checks ie pathways, and configuration checks. So I have added a discussion configuration in setup to make it work as expected.